llvm-build/llvm.ps1

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. (Join-Path $PSScriptRoot "utils.ps1")
+
+# Build
+
+$srcRoot = Get-RepoRoot
+$buildDir = Join-Path $srcRoot llvm-build
+$llvmCmakeFile = Join-Path $buildDir llvm.cmake
+$llvmRootDir = Join-Path $buildDir llvm-project
+$llvmBuildDir = Join-Path $llvmRootDir build
+$llvmDir = Join-Path $llvmRootDir llvm
+
+if (Test-CommandExists sccache) {
+    Write-Vso "Found sccache command"
+    # Set cap and make sure dir is created
+    $Env:SCCACHE_CACHE_SIZE = "2G"
+    Write-Vso "Starting sccache server"
+    & { sccache --start-server } -ErrorAction SilentlyContinue
+    Write-Vso "Started sccache server"
+}
+elseif (Test-CommandExists ccache) {
+    Write-Vso "Found ccache command"
+
+    if (-not (Test-Path env:\CCACHE_DIR)) {
+        # CCACHE_DIR needs to be set, get the value from ccache
+        $Env:CCACHE_DIR = exec { ccache -k cache_dir }
+    }
+    Assert (![string]::IsNullOrWhiteSpace($Env:CCACHE_DIR)) "CCACHE_DIR is not set"
+
+    # Set cap and make sure dir is created
+    if (!(Test-Path $Env:CCACHE_DIR)) {
+        mkdir $Env:CCACHE_DIR | Out-Null
+    }
```

Comment:
Looks like [the `-p`](https://man7.org/linux/man-pages/man1/mkdir.1.html) will not harm right after `mkdir`: "mkdir **-p** $Env:CCACHE_DIR".

Reply:
That's only true when `mkdir` resolves to the native command on a system, but on some systems and platforms `mkdir` is a function that wraps `New-Item -ItemType Directory`. The most portable option would be as follows:
```suggestion
    if (!(Test-Path $Env:CCACHE_DIR)) {
        New-Item -ItemType Directory -Force $Env:CCACHE_DIR | Out-Null
    }
```

Reply:
Cass is right, the explicit powershell way is the safest in this case. I'll update that in this file and in libllvm.ps1